src/ssl.c

```diff
@@ -17838,123 +17839,50 @@ size_t wolfSSL_get_client_random(const WOLFSSL* ssl, unsigned char* out,
        Returns WOLFSSL_SUCCESS on success, WOLFSSL_FAILURE on failure. */
     int wolfSSL_RAND_set_rand_method(const WOLFSSL_RAND_METHOD *methods)
     {
-        int ret = WOLFSSL_FAILURE;
-
-        if (initRandMethMutex == 0) {
-            if (wc_InitMutex(&rand_methods_mutex) != 0) {
-                WOLFSSL_MSG("Bad Init Mutex rand methods");
-                ret = BAD_MUTEX_E;
-            }
-            else {
-                initRandMethMutex = 1;
-            }
+    #ifndef WOLFSSL_NO_OPENSSL_RAND_CB
+        if (wolfSSL_RAND_InitMutex() == 0 && wc_LockMutex(&gRandMethodMutex) == 0) {
+            gRandMethods = methods;
+            wc_UnLockMutex(&gRandMethodMutex);
         }
```

Comment:
This function is now always returning WOLFSSL_SUCCESS, even if it fails.  We need an else case after this that returns WOLFSSL_FAILURE, it only succeeds if gRandMethods is set.  Ideally it would report if initializing or locking the mutex failed, but up to you if it's worth the extra lines.